samples/sensor/scd4x/src/main.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 Stephen Oliver
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/logging/log.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <stdio.h>
+
+#include <zephyr/drivers/sensor/scd4x.h>
+
+LOG_MODULE_REGISTER(main);
+
+#if !DT_HAS_COMPAT_STATUS_OKAY(sensirion_scd4x)
+#error "No sensirion,scd4x compatible node found in the device tree"
+#endif
+
+
+void main(void)
+{
+	printk("\n");
+
+	const struct device *scd = DEVICE_DT_GET_ANY(sensirion_scd4x);
+
+	struct sensor_value temp, hum, co2;
+
+	if (!device_is_ready(scd)) {
+		LOG_ERR("Device %s is not ready", scd->name);
+		return;
+	}
+
+	while (true) {
+		if (sensor_sample_fetch(scd)) {
+			LOG_ERR("Failed to fetch sample from SCD4X device");
```

Comment:
This line can be more meaningful if it prints the name of the device defined in the dts:

```c
LOG_ERR("Failed to fetch sample from %s device", scd->name);
```